src/sql/workbench/parts/restore/browser/restore.contribution.ts

```diff
@@ -60,7 +60,8 @@ MenuRegistry.appendMenuItem(MenuId.ObjectExplorerItemContext, {
 		id: OE_RESTORE_COMMAND_ID,
 		title: localize('backup', "Restore")
 	},
-	when: ContextKeyExpr.and(TreeNodeContextKey.NodeType.isEqualTo(NodeType.Database), ConnectionContextKey.Provider.isEqualTo(mssqlProviderName), ServerInfoContextKey.IsCloud.toNegated())
+	when: ContextKeyExpr.and(TreeNodeContextKey.NodeType.isEqualTo(NodeType.Database), ConnectionContextKey.Provider.isEqualTo(mssqlProviderName),
+		ServerInfoContextKey.IsCloud.toNegated(), MssqlNodeContext.IsSqlOnDemand.toNegated())
 });
```

Comment:
ServerInforContextKey instead of MssqlNodeContext here, and ItemContextKey bellow.